pkg/k8s/watchers/node.go

```diff
@@ -117,3 +135,109 @@ func (k *K8sWatcher) GetK8sNode(_ context.Context, nodeName string) (*v1.Node, e
 	}
 	return nodeInterface.(*v1.Node).DeepCopy(), nil
 }
+
+// CiliumNodeUpdater implements the subscriber.Node interface and is used
+// to keep CiliumNode objects in sync with the node ones.
+type CiliumNodeUpdater struct {
+	k8sWatcher    *K8sWatcher
+	nodeDiscovery *nodediscovery.NodeDiscovery
+}
+
+func NewCiliumNodeUpdater(k8sWatcher *K8sWatcher, nodeDiscovery *nodediscovery.NodeDiscovery) *CiliumNodeUpdater {
+	return &CiliumNodeUpdater{
+		k8sWatcher:    k8sWatcher,
+		nodeDiscovery: nodeDiscovery,
```

Comment:
[2]

I'm wondering if it would be better to scope down the access of `nodeDiscovery` from within this object. Can we potentially find an interface to abstract this away? This can also ease unit tests.

Reply:
So `nodeDiscovery` is used to:
* access the KV store: `nodeDiscovery.Registrar.UpdateLocalKeySync(k8sNodeParsed)`
* access the local node addresses: `nodeDiscovery.LocalNode.IPAddresses` (I'm still figuring out if I really need to access this or if I can just reuse the data I get from the CiliumNodeStore cache)